File: @LV1/1.3/spavanac/spavanac.cpp
#include <cstdio>

int main() {
    int a, b;
    scanf("%d %d", &a, &b);
    b -= 45;

    if (b < 0) {
        b += 60;
        a--;
    }

    if (a < 0) {
        a = 23;
    }

    printf("%d %d\n", a, b);
}

File: @LV1/1.3/spavanac/spavanac.go
package main

import "fmt"

func main() {
	var a, b int
	fmt.Scanf("%d %d", &a, &b)
	b -= 45

	if b < 0 {
		b += 60
		a--
	}

	if a < 0 {
		a = 23
	}

	fmt.Println(a, b)
}
